server/conf/evolutions/default/44.sql

```diff
@@ -0,0 +1,66 @@
+# --- Add database triggers to set the latest_status column on creating an application_event or
+# --- updating an application.
+# --- For rationale behind choosing database triggers, see:
+# ---   https://github.com/civiform/civiform/issues/3269#issuecomment-1232193163
+# --- Relevant documentation:
+# --- Trigger functions: https://www.postgresql.org/docs/current/sql-createtrigger.html
+# --- Create trigger statement: https://www.postgresql.org/docs/current/sql-createtrigger.html
+
+# --- !Ups
+
+alter table applications add column if not exists latest_status varchar;
+
+-- The latest status associated with an application is set to NULL if no application event entries
+-- exist. Otherwise, the text from the latest event corresponding to a status change is used.
+CREATE OR REPLACE FUNCTION retrieve_latest_status(app_id applications.id%TYPE) RETURNS text AS $$
```

Comment:
Functions are in a global namespace I assume so I think these names should be more specific to their usage.  Triggers too?

Here: retrieve_latest_application_status

next one: process_latests_status_on_application_change

etc

Reply:
Good call. Done.